vllm/config.py

```diff
@@ -678,6 +678,8 @@ def maybe_create_spec_config(
                 "Expected both speculative_model and "
                 "num_speculative_tokens to be provided, but found "
                 f"{speculative_model=} and {num_speculative_tokens=}.")
+        assert speculative_model is not None
+        assert num_speculative_tokens is not None
```

Comment:
i'm surprised you still need these while there's the if statement before. mypy should be catching these?

Reply:
I think the if above is `speculative_model is None and num_speculative_tokens is None`, so it doesn't cover only one of them is None.